peps/pep-0780.rst

```diff
@@ -0,0 +1,447 @@
+
+Specification
+=============
+
+The keywords "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**",
+"**SHALL NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**MAY**",
+and "**OPTIONAL**"" in this document are to be interpreted as described in
+:rfc:`RFC 2119 <2119>`.
+
+Concepts
+--------
+
+ABI features are a clear description of certain properties of the Python
+interpreter. While some of these features can be queried already today, they
+are not easily nor uniformly accessible.
+
+Hence, the proposed feature MUST be made available via the Python standard
+library as ``sys.abi_features``, and as the new environment marker variable
+``sys_abi_features``.
+
+ABI Features
+''''''''''''
+
+ABI features are intrinsic properties of the Python interpreter, expressed as
+simple, understandable strings.
+However, not all features are equally applicable to all Python interpreters or
+Python versions. For example, the distinction between free-threaded and
+GIL-enabled interpreters is only relevant for CPython 3.13 onwards, but the bitness
+of the interpreter is relevant for all interpreters.
+
+All interpreters MUST handle the following ABI features as stated.
+ABI features that are restricted to particular interpreters MUST NOT be
+provided by other interpreters.
+The features are subdivided into groups and for each group there MUST be
+exactly one feature present, except when the group is marked as optional, in
+which case there MUST be at most one feature present.
+
+``free-threading`` or ``gil-enabled`` (only CPython)
+    If the Python interpreter is free-threaded, the ``free-threading`` feature
+    MUST be present and the ``gil-enabled`` feature MUST NOT be present.
+    Otherwise, the ``gil-enabled`` feature MUST be present and the
+    ``free-threading`` feature MUST NOT be present.
+
+``debug`` (only CPython, optional)
+    This ABI feature is reserved for the ``--with-pydebug`` build of CPython.
+    If the interpreter is a CPython interpreter with ``Py_DEBUG`` capabilities,
+    the ``debug`` feature MUST be present.
+    On POSIX systems, this corresponds to ``"d" in sys.abiflags``.
+
+``32-bit`` or ``64-bit`` (optional)
+    The bitness of the interpreter, that is, whether it is a 32-bit or 64-bit
+    build [#bitness]_. If the bitness is unknown or neither 32-bit nor 64-bit,
+    this feature MUST NOT be present.
+
+
+Addition to the Python Standard Library
+'''''''''''''''''''''''''''''''''''''''
+
+Making the ABI features available in an easily accessible, expressive,
+standardized way is useful beyond the scope of environment markers.
+For example, ``"32-bit" in sys.abi_features`` is much more expressive than the
+current standard test of comparing ``sys.maxsize`` with  ``2**32``, which can
+be found more than ten thousand times on GitHub.
+If one wants to determine whether the interpreter is a debug build, there is
+currently no standardized, cross platform way to do so.
```

Comment:
Is it worth mentioning somewhere that e.g. `sys.abiflags` doesn't exist on Windows, but `sys.abi_features` will exist on every supported platform?

Reply:
I added a clause to that effect. Thanks!